recipes/ninja/1.10.x/patches/add_subproject_support.patch

```diff
@@ -0,0 +1,15 @@
+--- a/CMakeLists.txt	2020-07-16 14:12:55.679748344 +0200
++++ b/CMakeLists.txt	2020-07-16 14:13:44.435902524 +0200
+@@ -32,9 +32,9 @@
+ 			COMMAND ${RE2C} -b -i --no-generation-date -o ${OUT} ${IN}
+ 		)
+ 	endfunction()
+-	re2c(${CMAKE_SOURCE_DIR}/src/depfile_parser.in.cc ${CMAKE_BINARY_DIR}/depfile_parser.cc)
+-	re2c(${CMAKE_SOURCE_DIR}/src/lexer.in.cc ${CMAKE_BINARY_DIR}/lexer.cc)
+-	add_library(libninja-re2c OBJECT ${CMAKE_BINARY_DIR}/depfile_parser.cc ${CMAKE_BINARY_DIR}/lexer.cc)
++	re2c(${PROJECT_SOURCE_DIR}/src/depfile_parser.in.cc ${PROJECT_BINARY_DIR}/depfile_parser.cc)
++	re2c(${PROJECT_SOURCE_DIR}/src/lexer.in.cc ${PROJECT_BINARY_DIR}/lexer.cc)
```

Comment:
Does ninja use `re2c` which I'm adding in #2209

Reply:
I just checked and indeed.
It contains `find_program(RE2C re2c)`.
It does need to be added as a build requirement, unless you're modifying the lexer sources.